sig-and-wg/sig-core/proposals/asset-store-proposal.md

```diff
@@ -0,0 +1,274 @@
+- Update of the reference causes recreation of the assets in new location, removal from old location and update of the status and the references
+
+## Asset custom resource
+
+Asset resource mandatory information is the:
+- reference info about the source file/object/asset location that must be fetched by Asset controller with 3 different modes:
+  - `single` - the link points directly to asset that needs to be fetched
+  - `index` - the link to index.yaml file that contains reference to files that need to be separately fetch from a given relative location
+  - `package` - the link to the zip/tar file that must be unziped before it is uploaded
+```
+#sample of index file with markdown and assets files
+apiVersion: v1
+files:
+  - name: 01-overview.md
+    metadata:
+      title: MyOverview
+      type: Overview
+  - name: 02-details.md
+    metadata:
+      title: MyDetails
+      type: Details
+  - name: 03-installation.md
+    metadata:
+      title: MyInstallation
+      type: Tutorial
+  - name: assets/diagram.svg
+```
+- reference to the bucket where the asset should be stored
+```
+apiVersion: assetstore.kyma-project.io/v1alpha1
+kind: Asset
+metadata:
+  name: my-package-assets
+  namespace: default
+spec:
+  source:
+    mode: single # or index or package
+    url: https://some.domain.com/main.js
+  bucketRef:
+    name: my-bucket
+```
+
+The optional information is the:
+- ConfigMap reference, that points to the ConfigMap that introduces [a new file](https://github.com/kyma-project/kyma/blob/master/resources/core/charts/service-catalog-addons/charts/instances-ui/templates/configmap.yaml) that is also sent to the bucket along with other files
+- Asset validation webhook reference to a service that performs the validation of fetched assets before they are uploaded to the bucket. The use cases are:
+  - validation of specific file against some specification
+  - security validation
+```
+apiVersion: assetstore.kyma-project.io/v1alpha1
+kind: Asset
+metadata:
+  name: my-direct-assets
+  namespace: default
+spec:
+  source:
+    direct: https://some.domain.com/my.json
+    validationWebhookService:
+        name: swagger-operations-svc
+        namespace: default
+        endpoint: "/validate"
+    rewrites:
+      - keyvalue: 
+          basePath: /test/v2
+  bucketRef:
+    name: my-bucket
+status:
+  phase: Ready # Failed/Pending
+  reason: Validation failure
+  message: "file is not valid against provided json schema"
+```
+- Asset mutation webhook reference to a service that acts similar to the validation service mentioned above, but mutates the asset instead of just validating. The use case is for example the asset rewriting through `regex` operation or `keyvalue` (like for example modification in json specification)
```

Comment:
```suggestion
- An Asset mutation webhook reference to a service that acts similarly to the validation service mentioned above. The difference is that it mutates the asset instead of just validating it. The use case is, for example, the asset rewriting through the `regex` operation or `keyvalue`, or modification in the json specification).
```